src/raster/rasterband.rs

```diff
@@ -17,37 +17,49 @@ use ndarray::Array2;
 
 use crate::errors::*;
 
-/// Resampling algorithms, map GDAL defines
+/// Resampling algorithms used throughout various GDAL raster I/O operations.
+///
+/// # Example
+///
+/// ```rust
+/// use gdal::Dataset;
+/// # fn main() -> gdal::errors::Result<()> {
+/// use gdal::raster::ResampleAlg;
+/// let ds = Dataset::open("fixtures/tinymarble.tif")?;
+/// let band1 = ds.rasterband(1)?;
+/// let stats = band1.get_statistics(true, false)?.unwrap();
+/// // Down-sample a image using cubic-spline interpolation
+/// let buf = band1.read_as::<f64>((0, 0), ds.raster_size(), (2, 2), Some(ResampleAlg::CubicSpline))?;
+/// // In this particular image, resulting data should be close to the overall average.
+/// assert!(buf.data.iter().all(|c| (c - stats.mean).abs() < stats.std_dev / 2.0));
+/// # Ok(())
+/// # }
+/// ```
 #[derive(Debug, Copy, Clone)]
+#[repr(u32)]
```

Comment:
Not entirely sure about this value, or even if it should be specified.
`GDALRIOResampleAlg::Type` is `libc::c_uint`, which is `u32` on *nix. The [repr options](https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#repr-annotations-accepted-on-enums) maybe suggest `#[repr(C)]` should be used instead, and the right value will be selected based on the discriminant values? Advice appreicated.

Reply:
I don't think we need the `#[repr]` because of the cast in `to_gdal`.